streaming/java/streaming-runtime/src/main/java/org/ray/streaming/runtime/master/resourcemanager/ResourceManager.java

```diff
@@ -1,53 +1,19 @@
 package org.ray.streaming.runtime.master.resourcemanager;
 
-import java.util.List;
-import java.util.Map;
+import com.google.common.collect.ImmutableList;
 import org.ray.streaming.runtime.core.resource.Container;
-import org.ray.streaming.runtime.core.resource.Resources;
-import org.ray.streaming.runtime.master.scheduler.strategy.SlotAssignStrategy;
+import org.ray.streaming.runtime.master.resourcemanager.strategy.ResourceAssignStrategy;
 
 /**
- * The resource manager is responsible for resource de-/allocation and monitoring ray cluster.
+ * ResourceManager(RM) is responsible for resource de-/allocation and monitoring ray cluster.
  */
-public interface ResourceManager {
+public interface ResourceManager extends ResourceAssignStrategy {
 
   /**
-   * Get all registered container as a list.
+   * Get registered containers
    *
-   * @return A list of containers.
+   * @return the registered container list
    */
-  List<Container> getRegisteredContainers();
+  ImmutableList<Container> getRegisteredContainers();
```

Comment:
I found many interface methods return `ImmutableList/ImmutableMap` . It's necessary to emphasize it's immutable or just changed return type to `List/Map`

Reply:
It's more a taste

Reply:
Yes, comments have been added to related methods.